src/useSize/index.ts

```diff
@@ -0,0 +1,44 @@
+import React, { useEffect, useState, useRef, MutableRefObject, useLayoutEffect } from 'react';
+import ResizeObserver from 'resize-observer-polyfill';
+
+type args = HTMLElement | (() => HTMLElement);
+
+function useSizeRef<T = HTMLElement>(
+  ...args: args[]
+): [{ width: number; height: number }, MutableRefObject<T>] {
+  const element = useRef<T>(null);
+  const [state, setState] = useState({ width: 0, height: 0 });
+
+  useLayoutEffect(() => {
+    const hasPassedInRef = args.length === 1;
+    const passedInRef =
```

Comment:
```suggestion
    const passedInElement =
```
这里不是 `ref` 呀。。 `ref` 难道不是 `xxx.current` 的这种 `RefObject` 么。。（我是不是对 `ref` 有什么误解。。）

Reply:
这里命名改了